src/apps/backend/modules/account/internal/account_writer.py

```diff
@@ -87,3 +88,16 @@ def update_account_profile(*, account_id: str, params: UpdateAccountProfileParam
             raise AccountWithIdNotFoundError(id=account_id)
 
         return AccountUtil.convert_account_bson_to_account(updated_account)
+
+    @staticmethod
+    def delete_account(*, account_id: str) -> Account:
```

Comment:
Since account is deleted, it is not expected for this function return Account object